org.sonarlint.eclipse.core.tests/src/test/java/org/sonarlint/eclipse/core/internal/telemetry/SonarLintTelemetryTest.java

```diff
@@ -31,16 +44,34 @@
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.verifyNoMoreInteractions;
 import static org.mockito.Mockito.when;
+import static org.sonarlint.eclipse.core.internal.server.ServersManager.PREF_SERVERS;
 
-public class SonarLintTelemetryTest {
+public class SonarLintTelemetryTest extends SonarTestCase {
   private SonarLintTelemetry telemetry;
   private TelemetryManager engine = mock(TelemetryManager.class);
 
+  private static IProject project;
+  private static SonarLintProjectConfiguration configuration;
+  private static final IEclipsePreferences ROOT = InstanceScope.INSTANCE.getNode(SonarLintCorePlugin.PLUGIN_ID);
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+    project = importEclipseProject("SimpleProject");
+    // Configure the project
+    configuration = SonarLintCorePlugin.getInstance().getProjectConfigManager().load(new ProjectScope(project),
+        "A Project");
+  }
+
   @Before
   public void start() throws Exception {
     // Clear property that is set in the surefire arguments (see pom.xml)
     System.clearProperty(SonarLintTelemetry.DISABLE_PROPERTY_KEY);
     this.telemetry = createTelemetry();
+
+    project.open(MONITOR);
+    ROOT.node(PREF_SERVERS).removeNode();
+    SonarLintCorePlugin.getServersManager().stop();
+    SonarLintCorePlugin.getServersManager().init();
   }
 
   @After
```

Comment:
I know this is not part of your changes, but I wonder if there is not an issue here. Shouldn't we restore the 'disable' property to 'true' to ensure next test suites do not send ping to the real telemetry?

Reply:
I moved the check for the system property up in the call hierarchy, so that in the tests we focus on testing the real telemetry behavior. 
I am still not totally happy with the design (e.g. there are some null checks in different places which is a smell for me). I would suggest having two different implementations, e.g. SonarLintProductionTelemetry and another empty one (SonarLintOfflineTelemetry), that would better represent reality. I think we can keep that for later in the sprint, to first make progress on the remaining tickets. WDYT?

Reply:
I moved the check for the system property up in the call hierarchy, so that in the tests we focus on testing the real telemetry behavior.
I am still not totally happy with the design (e.g. there are some null checks in different places which is a smell for me). I would suggest having two different implementations, e.g. SonarLintProductionTelemetry and another empty one (SonarLintOfflineTelemetry), that would better represent reality. I think we can keep that for later in the sprint, to first make progress on the remaining tickets. WDYT?